### PR TITLE
SNOW-815559: Remove xfail from string type tests

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-815559

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/integ/scala/test_dataframe_suite.py [hunk 1]
```diff
@@ -1520,7 +1518,6 @@ def test_createDataFrame_with_given_schema(session):
             StructField("number", DecimalType(10, 3)),
             StructField("boolean", BooleanType()),
             StructField("binary", BinaryType()),
-            StructField("timestamp", TimestampType()),
             StructField("timestamp_ntz", TimestampType(TimestampTimeZone.NTZ)),
```
Comment (sfc-gh-aalam) [inline tests/integ/scala/test_dataframe_suite.py:1521]: Here setting timestamp with default timezone value currently maps to NTZ but can cause flakiness in the future so I removed it in favor of testing of other explicitly defined timezones.

**Comment by sfc-gh-aalam** [comment]
Pre-commit here: https://ci-dev-133.int.snowflakecomputing.com/job/RT-Precommit-Tests/12377/console

File: tests/integ/scala/test_dataframe_suite.py (hunk 1 above)
Comment (sfc-gh-sfan) [reply tests/integ/scala/test_dataframe_suite.py:1521]: Hmm but how will we cover the default use case?

**Review by sfc-gh-sfan — APPROVED** [review]

File: tests/integ/scala/test_dataframe_suite.py (hunk 1 above)
Comment (sfc-gh-aalam) [reply tests/integ/scala/test_dataframe_suite.py:1521]: I can try to use statement params to set default timezone to be ntz and run the test using that

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**